src/consensus/pbft/libbyz/prepare.cpp

```diff
@@ -41,7 +47,7 @@ Prepare::Prepare(View v, Seqno s, Digest& d, Principal* dst, bool is_signed) :
       signature(Digest d_, NodeId id_) : d(d_), id(id_) {}
     };
 
-    signature s(d, pbft::GlobalState::get_node().id());
+    signature s(d, rep().id);
```

Comment:
why this change? we want this node's id.

Reply:
Because we create `Prepare`s from the pre-prepare proofs in playback and we need to set the correct id otherwise the digest will be wrong